pkg/bpfdebug/trace.go

```diff
@@ -125,9 +125,9 @@ func (n *TraceNotify) traceSummary() string {
 
 // DumpInfo prints a summary of the trace messages.
 func (n *TraceNotify) DumpInfo(data []byte) {
-	fmt.Printf("%s flow %#x identity %d->%d state %s: %s\n",
+	fmt.Printf("%s flow %#x identity %d->%d state %s ifindex %s: %s\n",
 		n.traceSummary(), n.Hash, n.SrcLabel, n.DstLabel,
-		connState(n.Reason), GetConnectionSummary(data[TraceNotifyLen:]))
```

Comment:
There's going to be a lock/unlock for every single packet now. Won't this make the monitor slower and the connections slower as well?

Reply:
The bottleneck will be the gob decode/encode. The lock is absolutely fine here.